platform-sdk/base-apps/swirlds-base-sample/src/main/java/com/swirlds/base/sample/internal/AdapterHandler.java

```diff
@@ -0,0 +1,184 @@
+        long start = System.nanoTime();
+
+        final List<String> urlParts =
+                DataTransferUtils.urlToList(exchange.getRequestURI().toString());
+        final String id = urlParts.getLast();
+        final boolean isId = !id.equalsIgnoreCase(getControllerName());
+        Object result = null;
+        int statusCode;
+        try {
+            switch (requestMethod) {
+                case "GET":
+                    result = (isId && !id.contains("?"))
+                            ? retrieve(id)
+                            : retrieveAll(DataTransferUtils.getUrlParams(exchange));
+                    statusCode = 200;
+                    break;
+                case "POST":
+                    result = this.create(
+                            DataTransferUtils.deserializeJsonFromExchange(exchange, delegatedService.getResultType()));
+                    statusCode = 201;
+                    break;
+                case "PUT":
+                    if (!isId) {
+                        throw new IllegalArgumentException("Identification must be sent in url");
+                    }
+                    result = this.update(
+                            id,
+                            DataTransferUtils.deserializeJsonFromExchange(exchange, delegatedService.getResultType()));
+                    statusCode = 200;
+                    break;
+                case "DELETE":
+                    if (!isId) {
+                        throw new IllegalArgumentException("Identification must be sent in url");
+                    }
+                    this.delete(id);
+                    statusCode = 202;
+                    break;
+                case "OPTIONS": {
+                    handleOptionsRequest(exchange);
+                    return;
+                }
+
+                default:
+                    throw new UnsupportedOperationException("Operation not supported for " + requestMethod);
+            }
+        } catch (UnsupportedOperationException e) {
+            statusCode = 405;
+            result = Map.of("error", "Operation not supported for " + requestMethod);
+        } catch (IllegalArgumentException e) {
+            statusCode = 400;
+            result = Map.of("error", e.getMessage());
+        } catch (EntityNotFoundException e) {
+            statusCode = 404;
+            result = Map.of("error", e.getEntityType() + " with code " + e.getId() + " not found");
+        } catch (RuntimeException e) {
+            statusCode = 500;
+            result = Map.of("error", "Internal error");
+        }
+
+        final String response = DataTransferUtils.serializeToJson(result);
+        long duration = System.nanoTime() - start;
+        context.metrics().getOrCreate(ApplicationMetrics.REQUEST_TOTAL).increment();
+        context.metrics().getOrCreate(ApplicationMetrics.REQUEST_AVG_TIME).update(duration);
+        tps.count();
+        if (statusCode - 200 >= 100) {
```

Comment:
what is the difference to:
```suggestion
        if (statusCode >= 300) {
```
?

Reply:
Applied the suggestion